packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -39,17 +39,14 @@ export function readableStreamTee<T = any>(
   const writer2 = transformStream2.writable.getWriter()
 
   const reader = readable.getReader()
-  function read() {
-    reader.read().then(({ done, value }) => {
-      if (done) {
-        writer.close()
-        writer2.close()
-        return
-      }
-      writer.write(value)
-      writer2.write(value)
-      read()
-    })
+  async function read() {
+    const { done, value } = await reader.read()
+    if (done) {
+      await Promise.all([writer.close(), writer2.close()])
+      return
+    }
+    await Promise.all([writer.write(value), writer2.write(value)])
+    await read()
```

Comment:
Adding `await` to ensure the correctness of ordering. We might need to improve the error handling and messaging here as well.

Reply:
Did you capture a todo for this? Also writer2 vs writer2, can probably improve the names on these. Isn't immediately clear.